staging/src/k8s.io/legacy-cloud-providers/azure/azure.go

```diff
@@ -591,8 +594,9 @@ func (az *Cloud) configAzureClients(
 	az.LoadBalancerClient = loadbalancerclient.New(loadBalancerClientConfig)
 	az.SecurityGroupsClient = securitygroupclient.New(securityGroupClientConfig)
 	az.PublicIPAddressesClient = publicipclient.New(publicIPClientConfig)
-	// fileClient is not based on armclient, but it's still backoff retried.
-	az.FileClient = fileclient.NewAzureFileClient(&az.Environment, azClientConfig.Backoff)
+	az.FileClient, err = fileclient.NewAzureFileClient(&az.Environment, &az.AzureAuthConfig, az.SubscriptionID)
```

Comment:
why only fileclient needs return error here?

Reply:
I mean all other clients would not return error, it's strange only azure file client would return error, there are cases that user don't need azure file client.

Reply:
changed the function to `fileclient.New()` so that we can follow the style of others clients' `New()` func and reuse the code.